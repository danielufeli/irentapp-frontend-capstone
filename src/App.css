* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

.container {
  padding: 0;
  margin: 0;
}

.App {
  width: 100%;
  background-color: #fff;
}

/* main {
  width: 70%;
  display: flex;
  justify-content: center;
  flex-direction: row;
  align-items: center;
  padding: 20px;
}

.btn-login {
  font-size: 0.9rem;
  letter-spacing: 0.05rem;
  padding: 0.75rem 1rem;
}

.login_bg {
  background: url(/public/login_bg.jpg) no-repeat center center fixed;
  background-size: cover;
  min-height: 100%;
  overflow: hidden;
}

.btn-warning,
.btn-warning:hover,
.btn-warning:active,
.btn-warning:visited {
  background-color: #ffb701 !important;
}

.details {
  width: 800px;
  height: 100%;
}

.details .img-hold {
  width: 800px;
  height: 400px;
}

.details img {
  width: 100%;
}

.details .card-bod {
  width: 100%;
  padding: 10px;
}

.card-text {
  font-size: 15px;;
}

.card-text-money {
  font-weight: bold;
}

span {
  margin-right: 5px;
  font-weight: bold;
}

.btns {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  width: 40%;
  margin: 10px auto;
}

.same-size {
  width: 200px;
} */
