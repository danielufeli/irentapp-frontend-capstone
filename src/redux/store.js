import { configureStore } from '@reduxjs/toolkit';
import housesReducer from '../features/houses/housesState';

const store = configureStore({
  reducer: {
    houseList: housesReducer,
  },
});

export default store;
