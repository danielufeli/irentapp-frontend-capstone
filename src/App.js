import './App.css';
import { Routes, Route } from 'react-router-dom';
import Houses from './components/houses/houses';
import MyReservations from './components/reservations/MyReservations';
import AddReservation from './components/reservations/AddReservation';
import Login from './components/login/login';
import SignUp from './components/signup/signup';
import HouseForm from './components/houseForm/houseForm';
import HouseDetails from './components/houseDetails/houseDetails';
import HouseDelete from './components/houseDelete/houseDelete';

function App() {
  return (
    <div className="App">
      <Routes>
        <Route exact path="/" element={<Login />} />
        <Route exact path="/signup" element={<SignUp />} />
        <Route exact path="/houses" element={<Houses />} />
        <Route path="/reservations" element={<MyReservations />} />
        <Route path="/addreservation" element={<AddReservation />} />
        <Route path="/addHouse" element={<HouseForm />} />
        <Route path="houses/:houseId" element={<HouseDetails />} />
        <Route path="delete" element={<HouseDelete />} />
      </Routes>
    </div>
  );
}

export default App;
