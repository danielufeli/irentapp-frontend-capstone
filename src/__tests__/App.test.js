import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';
import store from '../redux/store';

describe('App', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
