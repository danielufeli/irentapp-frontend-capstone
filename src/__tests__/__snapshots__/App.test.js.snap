// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <section
          class="login_bg"
        >
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col-sm-9 col-md-7 col-lg-5 mx-auto"
              >
                <div
                  class="card border-0 shadow rounded-3 my-5"
                >
                  <div
                    class="card-body p-4 p-sm-5"
                  >
                    <h1
                      class="text-center"
                    >
                      Welcome!
                    </h1>
                    <h5
                      class="card-title text-center mb-5 fs-5"
                    >
                      Kindly Enter your Details to Login
                    </h5>
                    <br />
                    <form>
                      <div
                        class="form-group mb-3"
                      >
                        <input
                          class="form-control"
                          name="email"
                          placeholder="Enter your E-mail"
                          type="email"
                          value=""
                        />
                      </div>
                      <div
                        class="form-group mb-3"
                      >
                        <input
                          class="form-control"
                          name="password"
                          placeholder="Enter your Password"
                          type="password"
                          value=""
                        />
                      </div>
                      <div
                        class="form-check mb-3"
                      >
                        <label
                          class="form-check-label"
                          for="rememberPasswordCheck"
                        >
                          Remember password
                          <input
                            class="form-check-input"
                            id="rememberPasswordCheck"
                            type="checkbox"
                            value=""
                          />
                        </label>
                      </div>
                      <div
                        class="d-grid"
                      >
                        <button
                          class="btn btn-outline-dark btn-login text-uppercase fw-bold"
                          type="submit"
                        >
                          Sign in
                        </button>
                      </div>
                      <hr
                        class="my-4"
                      />
                      <div
                        class="d-grid mb-2"
                      >
                        <a
                          class="btn btn-warning btn-login text-uppercase fw-bold"
                          href="/signup"
                        >
                          Sign Up
                        </a>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <section
        class="login_bg"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-sm-9 col-md-7 col-lg-5 mx-auto"
            >
              <div
                class="card border-0 shadow rounded-3 my-5"
              >
                <div
                  class="card-body p-4 p-sm-5"
                >
                  <h1
                    class="text-center"
                  >
                    Welcome!
                  </h1>
                  <h5
                    class="card-title text-center mb-5 fs-5"
                  >
                    Kindly Enter your Details to Login
                  </h5>
                  <br />
                  <form>
                    <div
                      class="form-group mb-3"
                    >
                      <input
                        class="form-control"
                        name="email"
                        placeholder="Enter your E-mail"
                        type="email"
                        value=""
                      />
                    </div>
                    <div
                      class="form-group mb-3"
                    >
                      <input
                        class="form-control"
                        name="password"
                        placeholder="Enter your Password"
                        type="password"
                        value=""
                      />
                    </div>
                    <div
                      class="form-check mb-3"
                    >
                      <label
                        class="form-check-label"
                        for="rememberPasswordCheck"
                      >
                        Remember password
                        <input
                          class="form-check-input"
                          id="rememberPasswordCheck"
                          type="checkbox"
                          value=""
                        />
                      </label>
                    </div>
                    <div
                      class="d-grid"
                    >
                      <button
                        class="btn btn-outline-dark btn-login text-uppercase fw-bold"
                        type="submit"
                      >
                        Sign in
                      </button>
                    </div>
                    <hr
                      class="my-4"
                    />
                    <div
                      class="d-grid mb-2"
                    >
                      <a
                        class="btn btn-warning btn-login text-uppercase fw-bold"
                        href="/signup"
                      >
                        Sign Up
                      </a>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
